scipy/optimize/_linprog_ip.py

```diff
@@ -0,0 +1,1940 @@
+"""
+An interior-point method for linear programming.
+"""
+# Author: Matt Haberland
+
+from __future__ import print_function, division, absolute_import
+import numpy as np
+import scipy as sp
+import scipy.sparse as sps
+from warnings import warn
+from scipy.linalg import LinAlgError
+from .optimize import OptimizeResult, OptimizeWarning, _check_unknown_options
+#from scipy.optimize import _check_unknown_options
+from scipy.optimize._remove_redundancy import _remove_redundancy
+
+def _clean_inputs(
+        c,
+        A_ub=None,
+        b_ub=None,
+        A_eq=None,
+        b_eq=None,
+        bounds=None):
+    """
+    Given user inputs for a linear programming problem, return the
+    objective vector, upper bound constraints, equality constraints,
+    and simple bounds in a preferred format.
+
+    Parameters
+    ----------
+    c : array_like
+        Coefficients of the linear objective function to be minimized.
+    A_ub : array_like, optional
+        2-D array which, when matrix-multiplied by x, gives the values of the
+        upper-bound inequality constraints at x.
+    b_ub : array_like, optional
+        1-D array of values representing the upper-bound of each inequality
+        constraint (row) in A_ub.
+    A_eq : array_like, optional
+        2-D array which, when matrix-multiplied by x, gives the values of the
+        equality constraints at x.
+    b_eq : array_like, optional
+        1-D array of values representing the RHS of each equality constraint
+        (row) in A_eq.
+    bounds : sequence, optional
+        ``(min, max)`` pairs for each element in ``x``, defining
+        the bounds on that parameter. Use None for one of ``min`` or
+        ``max`` when there is no bound in that direction. By default
+        bounds are ``(0, None)`` (non-negative)
+        If a sequence containing a single tuple is provided, then ``min`` and
+        ``max`` will be applied to all variables in the problem.
+
+    Returns
+    -------
+    c : 1-D array
+        Coefficients of the linear objective function to be minimized.
+    A_ub : 2-D array
+        2-D array which, when matrix-multiplied by x, gives the values of the
+        upper-bound inequality constraints at x.
+    b_ub : 1-D array
+        1-D array of values representing the upper-bound of each inequality
+        constraint (row) in A_ub.
+    A_eq : 2-D array
+        2-D array which, when matrix-multiplied by x, gives the values of the
+        equality constraints at x.
+    b_eq : 1-D array
+        1-D array of values representing the RHS of each equality constraint
+        (row) in A_eq.
+    bounds : sequence of tuples
+        ``(min, max)`` pairs for each element in ``x``, defining
+        the bounds on that parameter. Use None for each of ``min`` or
+        ``max`` when there is no bound in that direction. By default
+        bounds are ``(0, None)`` (non-negative)
+
+    """
+
+    try:
+        if c is None:
```

Comment:
Okay, if it's that painful we can leave it alone. Why is this though, does it have to do with the benchmarks input? There must be something unusual about your inputs - pretty much any numpy or scipy function will do something weird if you call it with ``None`` as a first parameter, but there's basically zero chance of that happening.